File: java/src/checkout/GetAllowedCountries.java
package checkout;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.ultracart.admin.v2.CheckoutApi;
import com.ultracart.admin.v2.models.CheckoutAllowedCountriesResponse;
import com.ultracart.admin.v2.models.Country;
import com.ultracart.admin.v2.util.ApiException;
import common.Constants;

import java.util.List;

public class GetAllowedCountries {
    /**
     * Populates the country list with all countries the merchant has configured to accept
     * Reference Implementation: https://github.com/UltraCart/responsive_checkout
     */
    public static void execute() {
        CheckoutApi checkoutApi = new CheckoutApi(Constants.API_KEY);

        try {
            CheckoutAllowedCountriesResponse apiResponse = checkoutApi.getAllowedCountries();
            List<Country> allowedCountries = apiResponse.getCountries();

            Gson gson = new GsonBuilder().setPrettyPrinting().create();
            for (Country country : allowedCountries) {
                System.out.println(gson.toJson(country));
            }
        } catch (ApiException e) {
            e.printStackTrace();
        }
    }
}

File: ruby/checkout/get_allowed_countries.rb
# Reference Implementation: https://github.com/UltraCart/responsive_checkout
# A simple method for populating the country list boxes with all the countries this merchant has configured to accept.

require 'ultracart_api'
require_relative '../constants'

checkout_api = UltracartClient::CheckoutApi.new_using_api_key(Constants::API_KEY)

api_response = checkout_api.get_allowed_countries
allowed_countries = api_response.countries

allowed_countries.each do |country|
  puts country.inspect  # contains both iso2code and name
end